### ISSUE TITLE
ExcludedTypes for sampling doesn't include linked telemetry 

### ISSUE THREAD

**Issue by hallatore** [Microsoft/ApplicationInsights-dotnet#256]
When a data point is excluded from sampling it's linked data points are not.

We are currently testing it on a site which has sample set to 5% and we are excluding exceptions. This feature is ideal to keep the cost down while providing good logging of exceptions, and I think it will be a good configuration for many.

Current behavior: I get every exception, but not always the linked request/dependencies/etc.
Expected behavior: Get every exception and it's linked request/dependencies/etc.

Ref #243

**Comment by hallatore** [comment]
The issue I'm describing is very much dependent on Exception telemetry. It makes no sense if Dependency is the excluded telemetry.

The problem still remains, but maybe it could be resolved differently?

A good solution might be a way to tell the sampling to not sample an telemetry item. Then we can make a initializer that sets this value for stuff we don't want to sample. This would also have made it easy to create the ExcludeTypes behavior as an initializer.

**Comment by abaranch** [comment]
I did not quite got your question. So let me try to explain how it currently works.
If you did not provide ExludedTypes SamplingProcessor will use common sampling rate for all telemetry items. If one telemetry in determined to be "in" all related items are "in", if "out" - than "out".
Now we got a requirement from customers that they want either to have different sampling rates for different types or all events of certain types not sampled. So let's say we exclude exceptions from sampling. That means that you get all your exceptions. But other types are still sampled with default rates. Which means that some exception will be correlated with requests and some not, just because requests are sampled and exceptions are not. 
It feels that your would like to have a different scenario that does not fit in this enhancement: if request failed and it has exception, I want it "in". Currently it is not supported. For this scenario I think this may be a workaround: http://apmtips.com/blog/2015/05/06/diy-data-sampling/
